src/lib.rs

```diff
@@ -600,6 +603,28 @@ fn execute_instr(mut input: InstrContext) -> Option<InstrEffects> {
     sysvar_cache.fill_missing_entries(|pubkey, callbackback| {
         if let Some(account) = input.accounts.iter().find(|(key, _)| key == pubkey) {
             if account.1.lamports > 0 {
+                #[cfg(feature = "core-bpf")]
+                // BPF versions of programs, such as Address Lookup Table, rely
+                // on the new `SolGetSysvar` syscall. However, APIs for
+                // querying slot hashes built on top of `SolGetSysvar` are
+                // designed with the assumption that the `SlotHashes` data
+                // stored in the sysvar cache is `SlotHashes::size_of()` in
+                // length.
+                // See https://github.com/anza-xyz/agave/blob/96249691b4b7c873220b27376f271ead38392541/sdk/program/src/sysvar/slot_hashes.rs#L101.
+                //
+                // Fixtures may provide an incorrect sized slot hashes account,
+                // so this step is to rectify it by extending the buffer with
+                // all zeroes before adding it to the sysvar cache.
```

Comment:
we have an undocumented invariant that all sysvars should be rectified and valid before entering the harness, so you can probably just set a default value in the second `fill_missing_entries` call

Reply:
I think the trouble here is that when a sysvar is populated by account data using `fill_missing_entries`, the second call to `fill_missing_entries` won't override it since it's not `None`. I can double-check with a test, though.

Reply:
Some of the fixtures have valid data for `SlotHashes`, but they aren't `SlotHashes::size_of()`, so the API for `sol_get_sysvar` doesn't work since it expects the account to be size of `SlotHashes::size_of()`.

Many of them are just a few slot hashes, but the rest of the buffer needs to be zeroed out to the right size.

Reply:
ah gotcha, that makes sense, i see what this is doing now. ill give this a stamp